src/mscorlib/src/System/Collections/Generic/List.cs

```diff
@@ -1001,12 +1001,19 @@ public void Sort(IComparer<T> comparer)
             }
         }
 
-        // ToArray returns a new Object array containing the contents of the List.
+        // ToArray returns an array containing the contents of the List.
         // This requires copying the List, which is an O(n) operation.
         public T[] ToArray() {
             Contract.Ensures(Contract.Result<T[]>() != null);
             Contract.Ensures(Contract.Result<T[]>().Length == Count);
 
+#if FEATURE_CORECLR
+            if (_size == 0)
+            {
+                return Array.Empty<T>();
+            }
+#endif
+
             T[] array = new T[_size];
```

Comment:
Interesting idea.  Though in a case like this (and I expect others) you'd end up doing something like:
```C#
T[] array = Allocate(_size);
Array.Copy(_items, 0, array, 0, _size);
return array;
```
and you'd be doing a bit of extra work unnecessarily with the Array.Copy call.